clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsOnDemandCacheUpdater.groovy

```diff
@@ -57,13 +61,27 @@ class CatsOnDemandCacheUpdater implements OnDemandCacheUpdater {
   OnDemandCacheUpdater.OnDemandCacheStatus handle(OnDemandAgent.OnDemandType type, Collection<OnDemandAgent> onDemandAgents, Map<String, ? extends Object> data) {
     boolean hasOnDemandResults = false
     for (OnDemandAgent agent : onDemandAgents) {
+      boolean atomicScheduler = false;
+      AgentScheduler scheduler = null
+      if (agent instanceof AgentSchedulerAware) {
+        scheduler = agent.agentScheduler
```

Comment:
Yeah there is only one scheduler at a time, but I figured that's an implicit assumption and wanted to avoid adding code that will break in weird ways if two are running at a time. WDYT?